pint/diskcache.py

```diff
@@ -0,0 +1,121 @@
+"""
+    pint.diskmemo
+    ~~~~~~~~~~~~~
+
+    Functions for disk memoization.
+
+    To avoid global objects, a DiskCache folder must be
+    created where the cache folder can be specified.
+    Files are stored  under a name generated
+    by hashing with sha1 the raw filesystem path as encoded by
+    fs.encode().
+
+    :copyright: 2022 by Pint Authors, see AUTHORS for more details.
+    :license: BSD, see LICENSE for more details.
+"""
+
+from __future__ import annotations
+
+import hashlib
+import pathlib
+import pickle
+from typing import Union
+
+
+class DiskCache:
+    """
+
+    Parameters
+    ----------
+    cache_folder
+        indicates where the cache files will be saved.
+    extra_hash_info
+        extra information that will be used to create the hashed filename
+        Can be used, for example, to make the filename depend on the
+        Python version.
+    """
+
+    def __init__(
+        self, cache_folder: Union[str, pathlib.Path], extra_hash_info: tuple = ()
+    ):
+        self.cache_folder = pathlib.Path(cache_folder)
+        self.cache_folder.mkdir(parents=True, exist_ok=True)
+        self._extra_hash_info = extra_hash_info
+
+    def cachepath_for(self, obj) -> pathlib.Path:
+        """Generate a Path representing the location of a memoized file
+        for a given filepath or object.
+        """
+        if isinstance(obj, pathlib.Path):
+            hd = hashlib.sha1(bytes(obj.resolve()))
```

Comment:
We should rely on File hashing there instead of hashing the file path + relying on datetime.

So instead use : obj.read_bytes()

I would get rid of the code using `stats` & modified time, code will be simpler & safer.